swap_meet/vendor.py

```diff
@@ -1,2 +1,87 @@
 class Vendor:
-    pass
\ No newline at end of file
+    '''
+    A class to represent a person.
+    '''
+    
+    def __init__(self, inventory=None):
+        '''Construct inventory attribute for vendor. Value of attribute is a list.
+        '''
+        self.inventory = inventory if inventory else []
+
+    def add(self, new_item):
+        '''
+        add new item to person's inventory, return item
+        '''
+        self.inventory.append(new_item)
+        return new_item
+
+    def remove(self, item_to_remove):
+        '''
+        remove item from inventory, return item
+        '''
+        if item_to_remove not in self.inventory:
+            return False
+        self.inventory.remove(item_to_remove)
+        return item_to_remove
+
+    def get_by_category(self, category):
+        '''
+        make a list of given category and return it
+        '''
+        category_list = [item for item in self.inventory if item.category == category]
+        return category_list
+
+    def swap_items(self, friend, my_item, their_item):
+        '''swap items from vendor inventory and friend inventory. Return True.
+            If there are no items in vendor or friend inventory -> return False
+        '''
+        if my_item not in self.inventory or their_item not in friend.inventory:
+            return False
+
+        self.add(friend.remove(their_item))
+        friend.add(self.remove(my_item))
```

Comment:
I think I steered y'all wrong with `remove` vs `pop`. I thought `remove` returned the item it removed, but only `pop` does that. So what is being appended is actually `None`, not the item. Our test isn't checking for that unfortunately.

Sorry for that confusion. Something like this should work:

```suggestion
        self.add(their_item)
        friend.add(my_item)
        friend.remove(their_item)
        self.remove(my_item)
```